### PR TITLE
storage/engine: add get_cf for snapshot.

### PR DESCRIPTION
**Pull request by disksing**
@BusyJay @siddontang @ngaut

### CODE REVIEW

File: src/storage/engine/rocksdb.rs [hunk 1]
```diff
@@ -181,20 +175,48 @@ impl<'a> Cursor for DBIterator<'a> {
...
 
     fn iter<'b>(&'b self) -> Result<Box<Cursor + 'b>> {
         trace!("RocksSnapshot: create iterator");
-        Ok(box RocksSnapshot::iter(self))
+        Ok(box DBSnapshot::iter(&self.snap))
     }
 }
 
+fn get_cf_handle<'a>(db: &'a DB, cf: CfName) -> Result<&'a DBCFHandle> {
+    db.cf_handle(cf).ok_or(RocksDBError::new("cf not found.".to_string()).into_engine_error())
```
Comment (BusyJay) [inline src/storage/engine/rocksdb.rs]: `ok_or_else`

**Comment by siddontang** [comment]
any test?

File: src/storage/engine/mod.rs [hunk 2]
```diff
@@ -67,6 +67,7 @@ pub trait Engine: Send + Sync + Debug {
 
 pub trait Snapshot {
     fn get(&self, key: &Key) -> Result<Option<Value>>;
+    fn get_cf(&self, cf: CfName, key: &Key) -> Result<Option<Value>>;
```
Comment (zhangjinpeng1987) [inline src/storage/engine/mod.rs]: cf_get() maybe better

Comment (siddontang) [inline src/storage/engine/mod.rs]: The rust rocksdb uses the same name.

File: src/storage/engine/rocksdb.rs (hunk 1 above)
Comment (disksing) [inline src/storage/engine/rocksdb.rs]: @BusyJay What is the different between them?

**Comment by disksing** [comment]
@siddontang After this PR rocksdb engine's reads use a snapshot, so I think previous tests should cover it.

File: src/storage/engine/rocksdb.rs [hunk 3]
```diff
@@ -181,20 +175,49 @@ impl<'a> Cursor for DBIterator<'a> {
...
     }
 
     fn iter<'b>(&'b self) -> Result<Box<Cursor + 'b>> {
         trace!("RocksSnapshot: create iterator");
-        Ok(box RocksSnapshot::iter(self))
+        Ok(box DBSnapshot::iter(&self.snap))
     }
 }
 
+fn get_cf_handle<'a>(db: &'a DB, cf: CfName) -> Result<&'a DBCFHandle> {
```
Comment (BusyJay) [inline src/storage/engine/rocksdb.rs]: I don't think lifetime here is necessary.

Comment (disksing) [inline src/storage/engine/rocksdb.rs]: @BusyJay I don't know the detailed reason but it won't compile without `'a`.

**Comment by BusyJay** [comment]
LGTM

**Comment by siddontang** [comment]
LGTM

**Pull request merged**